src/agoradatatools/etl/transform/model_overview.py

```diff
@@ -0,0 +1,126 @@
+"""
+This module contains the transformation logic for the model_overview datasets.
+This is for the Model AD project.
+"""
+import pandas as pd
+from typing import Any, Dict, List
+
+from agoradatatools.etl.utils import (
+    check_required_datasets_and_columns,
+    remove_duplicates_keep_order,
+)
+from agoradatatools.etl.transform.model_details import process_genetic_info
+
+REQUIRED_INPUT = {
+    "model_info": [
+        "model",
+        "matched_controls",
+        "model_type",
+        "contributing_group",
+        "study_synid",
+        "rrid",
+        "jax_id",
+        "alzforum_id",
+        "genotype",
+        "aliases",
+    ],
+    "model_results_info": [
+        "model",
+        "gene_expression",
+        "disease_correlation",
+        "pathology",
+        "biomarkers",
+    ],
+    "allele_info": [
+        "model",
+        "modified_gene",
+        "gene_ensembl_id",
+        "allele",
+        "allele_type",
+        "mgi_allele_id",
+    ],
+    "human_transgene_allele_map": [
+        "mgi_allele_id",
+        "gene_symbol",
+        "human_ensembl_id",
+    ],
+}
+
+
+def transform_model_overview(
+    datasets: Dict[str, pd.DataFrame],
+    required_input: Dict[str, List[str]] = REQUIRED_INPUT,
+) -> List[Dict[str, Any]]:
+    """
+    Transforms the model_overview source files into a structured format for Model AD.
```

Comment:
Can you expand on this comment? It needs an `Arguments` and `Returns` but also maybe describe what the output format is supposed to be.